apps/wpcom-block-editor/src/wpcom/features/override-edit-site-back-button.js

```diff
@@ -0,0 +1,36 @@
+import { __experimentalNavigationBackButton as NavigationBackButton } from '@wordpress/components';
+import domReady from '@wordpress/dom-ready';
+import { createElement } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { registerPlugin } from '@wordpress/plugins';
+/**
+ * Conditional dependency.  We cannot use the standard 'import' since this package is
+ * not available in the post editor and causes WSOD in that case.  Instead, we can
+ * define it from 'require' and conditionally check if it is available for use.
+ */
+const editSitePackage = require( '@wordpress/edit-site' );
+
+/**
+ * Ensures that the site editor's back to dashboard button always leads back to Calypso from
+ * both wp-admin and the calypso iframe.
+ */
+function overrideBackToDashboardButton() {
+	if ( ! editSitePackage?.__experimentalMainDashboardButton ) return;
+
+	const SiteEditorDashboardFill = editSitePackage.__experimentalMainDashboardButton;
+	const siteSlug = window.location.hostname;
```

Comment:
Unsure if there's a better way to retrieve the site slug from the scope of the `wpcom-block-editor` plugin 🤔

Reply:
I think it is fine to use `window.location` to retrieve the site slug for this solution.